### PR TITLE
Report errors in path canonicalizer

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
Ah, this finally reports error in assembly reference resolution:
```
error: symbol not found in `::`: `a`
  ┌─ test.asm:3:9
  │
3 │ let t = a + b;
  │         ^

```
Previously, there was no indication about where the symbol `a` is referenced.

File: pil-analyzer/tests/types.rs [hunk 1]
```diff
@@ -156,6 +156,8 @@ fn generic_fixes_concrete() {
     // but in order to do that, we have to unify the derived type of
     // `y` with the declared and this would maybe create problems in that
     // we would not derive the most generic type for generic functions.
+
+    // TODO revisit this test, check if we are missing an error.
```
Comment (leonardoalt) [inline pil-analyzer/tests/types.rs:160]: when?

Comment (chriseth) [reply pil-analyzer/tests/types.rs:160]: Before merging.

Comment (chriseth) [reply pil-analyzer/tests/types.rs]: Ah, it was already merged.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**